pkg/clustermanager/cluster_manager.go

```diff
@@ -833,6 +838,21 @@ func (c *ClusterManager) waitForAllControlPlanes(ctx context.Context, cluster *t
 	return nil
 }
 
+func (c *ClusterManager) removeOldMachineDeployments(ctx context.Context, workloadCluster *types.Cluster, provider providers.Provider, currentSpec, newSpec *cluster.Spec) error {
+	workerConfigs := provider.BuildMapForWorkerNodeGroupsByName(newSpec.Spec.WorkerNodeGroupConfigurations)
+	for _, prevWorkerNodeGroupConfig := range currentSpec.Spec.WorkerNodeGroupConfigurations {
+		if _, ok := workerConfigs[prevWorkerNodeGroupConfig.Name]; !ok {
+			workerNodeGroupName := fmt.Sprintf("%s-%s", workloadCluster.Name, prevWorkerNodeGroupConfig.Name)
+			err := c.clusterClient.DeleteMachineDeployment(ctx, workloadCluster.KubeconfigFile, workerNodeGroupName)
```

Comment:
Does CAPI take care of deleting the `KubeadmConfigTemplate` and the provider machine template?

Reply:
No it took care of deleting the nodes & machine deployments, but it left the old resources in the other two. Updated the method to delete all resources related to unwanted node group.